sessiontxn/interface.go

```diff
@@ -120,7 +122,7 @@ type TxnContextProvider interface {
 	// OnStmtRetry is the hook that should be called when a statement is retried internally.
 	OnStmtRetry(ctx context.Context) error
 	// Advise is used to give advice to provider
-	Advise(tp AdviceType) error
+	Advise(tp AdviceType, val []any) error
```

Comment:
Try to avoid using `any` type but a concrete type in the interface definition, it would make it difficult to use and maintain.

Reply:
It depends on the design of what's planned to be passed inside and how the parameters would be used. The most commonly used way is to abstract some `interface` or `trait` to describe the meaning of the input parameters, for example, you could abstract a new type named `TxnAdvice` which contains both advice type and value, or abstract a new interface which defines what should be passed in as the `advice` parameters.  
The main thing is to answer `what's an advice and how to describe it, what's the design?`, but not just leave something like `whatever type is acceptable, perhaps some will work some would not`.

In general, don't use the `any` or `inteface{}` types it's not maintainable and other people have no idea what should be passed or how to build another `advice` type.